reference/test/matrix/diagonal_kernels.cpp

```diff
@@ -388,4 +413,46 @@ TYPED_TEST(DiagonalComplex, MtxIsConjugateTransposable)
 }
 
 
+TYPED_TEST(DiagonalComplex, InplaceAbsolute)
+{
+    using Diag = typename TestFixture::Diag;
+    using value_type = typename TestFixture::value_type;
+
+    auto exec = gko::ReferenceExecutor::create();
+    auto diag = Diag::create(exec, 3);
+    auto diag_values = diag->get_values();
+    diag_values[0] = value_type{3.0, -4.0};
+    diag_values[1] = value_type{-3.0, 0.0};
+    diag_values[2] = value_type{0.0, -1.5};
+
+    diag->compute_absolute_inplace();
+
+    EXPECT_EQ(diag_values[0], (value_type{5.0, 0.0}));
+    EXPECT_EQ(diag_values[1], (value_type{3.0, 0.0}));
+    EXPECT_EQ(diag_values[2], (value_type{1.5, 0.0}));
+}
+
+
+TYPED_TEST(DiagonalComplex, OutplaceAbsolute)
+{
+    using Diag = typename TestFixture::Diag;
+    using value_type = typename TestFixture::value_type;
+    using abs_type = gko::remove_complex<value_type>;
+
+    auto exec = gko::ReferenceExecutor::create();
```

Comment:
nit: newline too much:
```suggestion
    using abs_type = gko::remove_complex<value_type>;
    auto exec = gko::ReferenceExecutor::create();
```